pkg/apis/app/v1alpha1/kogitoinfra_types.go

```diff
@@ -19,45 +19,49 @@ import (
 }
 
 // KogitoInfraSpec defines the desired state of KogitoInfra.
 // +k8s:openapi-gen=true
 type KogitoInfraSpec struct {
 	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
 
-	// Indicates if Infinispan should be installed or not using Infinispan Operator.
-	// Please note that the Infinispan Operator must be installed manually on environments that doesn't have OLM installed.
-	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
-	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Install Infinispan"
-	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:booleanSwitch"
-	InstallInfinispan bool `json:"installInfinispan,omitempty"`
-	// Indicates if Kafka should be installed or not using Strimzi (Kafka Operator).
-	// Please note that the Strimzi must be installed manually on environments that doesn't have OLM installed.
+	// +optional
+	// Resource for the service. Example: Infinispan/Kafka/Keycloak.
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
-	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Install Kafka"
-	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:booleanSwitch"
-	InstallKafka bool `json:"installKafka,omitempty"`
-	// Whether or not to install Keycloak using Keycloak Operator.
-	// Please note that the Keycloak Operator must be installed manually on environments that doesn't have OLM installed.
-	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
-	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Install Keycloak"
-	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:booleanSwitch"
-	InstallKeycloak bool `json:"installKeycloak,omitempty"`
+	Resource Resource `json:"resource,omitempty"`
 }
 
 // KogitoInfraStatus defines the observed state of KogitoInfra.
 // +k8s:openapi-gen=true
 type KogitoInfraStatus struct {
-	Condition  KogitoInfraCondition            `json:"condition,omitempty"`
-	Infinispan InfinispanInstallStatus         `json:"infinispan,omitempty"`
-	Kafka      InfraComponentInstallStatusType `json:"kafka,omitempty"`
-	Keycloak   InfraComponentInstallStatusType `json:"keycloak,omitempty"`
+	Condition KogitoInfraCondition `json:"condition,omitempty"`
+
+	// +optional
+	// +mapType=atomic
+	// Application properties to be added to the runtime container.
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	AppProps map[string]string `json:"appProps,omitempty"`
+
+	// +optional
+	// +listType=atomic
+	// Environment variables to be added to the runtime container.
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	EnvVars []v1.EnvVar `json:"envVars,omitempty"`
```

Comment:
```suggestion
	// Environment variables extracted from the linked resource that will be added to the deployed Kogito service.
	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
	Env []v1.EnvVar `json:"env,omitempty"`
```

My suggestion to change to `Env` is to have coherence with #586

Reply:
done